service_manager_test.go

```diff
@@ -153,41 +172,46 @@ func TestServiceManagerStartWithDependency(t *testing.T) {
 		t.Fatal("can not init service manager: ", err)
 	}
 
-	recorded := []ServiceMessage{}
 	m.Start("B")
+
+	recorded := make([]ServiceMessage, 0, len(messages))
+
 	for message := range messages {
 		switch message.Name {
 		case "A":
 			if message.Type == MessageState && message.State == StateRunning {
-				atomic.AddInt64(&aStarts, 1)
-
+				atomic.AddInt64(&aStarts, aStarted)
```

Comment:
This is more obscure that it originally was. 
It was a common pattern "increment"